Formula/mongodb-csfle.rb

```diff
@@ -0,0 +1,23 @@
+class MongodbCsfle < Formula
+  desc "MongoDB csfle shared library for Client Side Encryption"
+  homepage "https://www.mongodb.com/"
+
+  # TODO(MONGOSH-1193): Bump to production version after 6.0 is released
+  url "https://downloads.mongodb.com/osx/mongo_csfle_v1-macos-x86_64-enterprise-6.0.0-rc0.tgz"
```

Comment:
I think we need to account for x86_64 vs arm64 here, since we now have native builds for macOS ARM.

Reply:
@acmorrow That applies to all the formulas, correct? I don’t know how to do this properly, do you happen to?

Reply:
@acmorrow I dug around and couldn't find any way to do this with the way we're publishing the formulas here. Homebrew generally expects `url` to point to a source tarball, and we just don't do that.

I'd honestly just add separate formulas for arm64. That might not be pretty, but it would work well enough, I assume. How does that sound?

Reply:
I'm not quite sure how that would work. Can you elaborate a bit on what you had in mind? Like we would have a `Formula/mongodb-csfle.rb` and a `Formula/mongodb-csfle-arm64.rb`? How would homebrew know to use the arm one?

Reply:
I think maybe we can interrogate `Hardware.CPU.arm` (see https://github.com/Homebrew/brew/blob/master/Library/Homebrew/hardware.rb#L147) and set `url` difrerently.

Reply:
Yeah, done as you suggested :slightly_smiling_face: